std/datetime.d

```diff
@@ -29161,6 +29191,72 @@ private:
         _hasDST = hasDST;
     }
 
+    // Android concatenates the usual timezone directories into a single file,
+    // tzdata, along with an index to jump to each timezone's offset.  In older
+    // versions of Android, the index was stored in a separate file, zoneinfo.idx,
+    // whereas now it's stored at the beginning of tzdata.
+    version(Android)
+    {
+        // Keep track of whether there's a separate index, zoneinfo.idx
+        static bool separate_index;
+
+        // Extracts the name of each time zone and the offset where its data is
+        // located in the tzdata file from the index and caches it for later.
+        static uint[string] tzdataIndex(string tzDir)
+        {
+            import std.format : format;
+            import std.path : asNormalizedPath, chainPath;
+            import std.conv : to;
+
+            static uint[string] _tzIndex;
```

Comment:
Oh I see. One way of dealing with this is to initialize it in a module constructor rather than lazily, which means it can be immutable. I don't know how/if `shared` works with AAs.